models/release.go

```diff
@@ -41,7 +41,7 @@ type Release struct {
 	CreatedUnix      util.TimeStamp `xorm:"INDEX"`
 }
 
-func (r *Release) loadAttributes(e Engine) error {
+func (r *Release) loadAttributes() error {
 	var err error
 	if r.Repo == nil {
 		r.Repo, err = GetRepositoryByID(r.RepoID)
```

Comment:
This should take the engine, i.e. be getRepositoryByID(e, r.RepoID)

Reply:
And line 53 should similarly use the provided engine

Reply:
and line 58 should be getReleaseAttachments(e, r) assuming that exists...

Reply:
Ah ok. Although that is nothing I changed, but that's why I removed the `Engine` since it wasn't used at all.

Reply:
Fixed.

Reply:
`getReleaseAttachments` does not exist though.

Reply:
I think it needs to exist - if it doesn't any use of loadAttributes() within a transaction could cause a deadlock

Reply:
(Sorry it's not the fault of this PR but it's worth fixing)

Reply:
Since this kind of is a refactoring pr, I thinks it's ok to fix smaller things like this when we stumble upon them, but I'd prefer if these won't get to be the majority of things this pr fixes. So as long as there aren't too many, I'll fix these. Though we should probably do a seperate pr to specifically fix things like this (seperatly).

(I've fixed the thing with `GetReleaseAttachments`)